vector_index.go

```diff
@@ -29,6 +29,12 @@ func init() {
 	reflectStaticSizeVectorDoc = int(reflect.TypeOf(vd).Size())
 }
 
+type VectorIndex interface {
```

Comment:
Also, a better position for this would be within scorch_segment_api.

Reply:
Not sure I follow.
As I understand, scorch_segment_api has segment specific constructs. Hence, I have added search related constructs in segment_vector.go there but I think this is better off here.

Reply:
I disagree, this interface is not needed anywhere here and no other construct that deploys an interface declared in bleve_index_api calls this directly.

A VectorIndex, as I see it - is an interpretation of content that reside within a Segment and so this is better off somewhere else. On second thought - you can even declare this in section.go directly within zapx.

bleve_index_api and scorch_segment_api are for interfaces and constructs that are used across bleve and zapx to not have to introduce a cyclic dependency.

Reply:
can't declare this within segment.go in zapx since scorch_segment_api requires it in the VectorSegment interface.
Hence, moved it to scorch_segment_api.